### ISSUE TITLE
Linux SocketCAN structs and constants (attempt #2)

### ISSUE THREAD

**Issue by fpagliughi** [jbaublitz/neli#206]
Big apologies for dropping the ball on #140 a while back. Work on the socketcan crate stalled while the project maintainership was being ironed out, but now a lot of progress is being made and one of the last major aspects to be added for the upcoming release is Netlink access to query and update the CANbus network interfaces.

It appears that most of what we need added is contained in the `can/netlink.h` header as can be seen here:
https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/can/netlink.h

It doesn't look too crazy; a few constants, bitflags, and structs. The things I'm still uncertain about is which parts of it might go here in the `neli` crate, which in `libc`, and what is best to just drop into the `socketcan` crate.

Any guidance would be appreciated.

**Comment by fpagliughi** [comment]
BTW... we've been discussing this work here:
https://github.com/socketcan-rs/socketcan-rs/issues/32

**Comment by jbaublitz** [comment]
Hi @fpagliughi! So what I'd recommend is first creating a PR against libc with all of the constants that you want added. Once you've done that and it's been merged and released in a new version, let me know which constants you want added to neli and the version of libc that contains them. From there, I'll put up a PR and put out a new version with those included. Sound good?

**Comment by jbaublitz** [comment]
Hi @fpagliughi, any way I can help with this? Or is this kind of a background task?

**Comment by fpagliughi** [comment]
Hey @jbaublitz. I was having trouble figuring out how to integrate a few things, but eventually found `#[neli_enum(...)]` and some related items, and was able to get everything working together from my crate. I put out a new release a few weeks back.
(Really nice job on your end making it so easy).

But now I do have a submodule of structs and consts [here](https://github.com/socketcan-rs/socketcan-rs/blob/master/src/nl/rt.rs) that really should be pushed upstream to libc and neli. I have to circle through a couple of other open-source projects that need to get out, then I will get back to this. Hopefully in a few weeks.

**Comment by fpagliughi** [comment]
Yeah, thanks @jbaublitz . Will do. Definitely want to get this cleaned up when time allows.
